### PR TITLE
chore: Port `FrameworkElementAutomationPeer`

### PR DESCRIPTION
**Pull request by morning4coffe-dev**
GitHub Issue (If applicable): part of #14344 

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

The implementation of `FrameworkElementAutomationPeer` is currently incomplete. 

## What is the new behavior?

Porting the implementation from winui3/release/1.5-stable open source code.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/DirectUI/AccessKeyStringBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+// MUX Reference AccessKeyStringBuilder.cpp, tag winui3/release/1.5-stable
+
+using System;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Automation.Peers;
+using Microsoft.UI.Xaml.Documents;
+
+namespace Uno.UI.DirectUI;
```
Comment (MartinZikmund) [inline src/Uno.UI/DirectUI/AccessKeyStringBuilder.cs:10]: Use `DirectUI` namespace

File: src/Uno.UI/DirectUI/ScrollItemAdapter.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+// MUX Reference ScrollItemAdapter_Partial.cpp, tag winui3/release/1.5-stable
+
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Automation.Peers;
+
+namespace Uno.UI.DirectUI;
```
Comment (MartinZikmund) [inline src/Uno.UI/DirectUI/ScrollItemAdapter.cs:8]: Use `DirectUI` namespace

File: src/Uno.UI/DirectUI/AccessKeyStringBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+// MUX Reference AccessKeyStringBuilder.cpp, tag winui3/release/1.5-stable
```
Comment (MartinZikmund) [inline src/Uno.UI/DirectUI/AccessKeyStringBuilder.cs:3]: The tag should be something like `1.5.x-stable`:

image

File: src/Uno.UI/UI/Xaml/Automation/Peers/FrameworkElementAutomationPeer.cs [hunk 4]
```diff
@@ -18,200 +27,575 @@
...
-			return true;
+			// this condition checks that if Control is visible and if it's popup then it must be open
+			childIsAcceptable = isPopupOpen && value == Visibility.Visible;
 		}
 
-		protected override void SetFocusCore()
+		return childIsAcceptable;
+	}
+
+	public IReadOnlyList<AutomationPeer> GetAutomationPeerCollection(DependencyProperty eProperty)
```
Comment (MartinZikmund) [inline src/Uno.UI/UI/Xaml/Automation/Peers/FrameworkElementAutomationPeer.cs:535]: Make sure only the public members from WinUI are also public on our side -  
image

**Review by MartinZikmund — APPROVED** [review]

File: src/Uno.UI/DirectUI/AccessKeyStringBuilder.cs (hunk 3 above)
Comment (morning4coffe-dev) [reply src/Uno.UI/DirectUI/AccessKeyStringBuilder.cs]: Ah, I see. I marked the branch, not the tag, thanks.

**Pull request merged**